compiler/src/main/scala/org/scalajs/core/compiler/PrepJSInterop.scala

```diff
@@ -1027,6 +1031,17 @@ abstract class PrepJSInterop extends plugins.PluginComponent
       ddef
     }
   }
+
+  private def checkInternalAnnotations(sym: Symbol): Unit = {
+    for {
+      ann <- compilerInternalAnnotations
+      annSym <- sym.getAnnotation(ann)
+      if annSym.pos.isDefined
```

Comment:
> Unfortunately in PreTyperComponent the name is still not fully qualified. 

Of course not. It needs the typer to find out the fully qualified name. But if the typer doesn't change the trees, you know that it is not a tree generated by the compiler if it is not fully qualified.